src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/BulkUpdateStudyInstance.sql

```diff
@@ -0,0 +1,157 @@
+﻿/*************************************************************
+    Stored procedures for updating an instance status.
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     BulkUpdateStudyInstance
+--
+-- DESCRIPTION
+--     Bulk update all instances in a study, and update extendedquerytag with new watermark.
+--
+-- PARAMETERS
+--     @batchSize
+--         * The desired number of instances per batch. Actual number may be smaller.
+--     @partitionKey
+--         * The partition key.
+--     @studyInstanceUid
+--         * The study instance UID.
+--     @patientId
+--         * The Id of the patient.
+--     @patientName
+--         * The name of the patient.
+--     @referringPhysicianName
+--         * The referring physician name.
+--     @studyDate
+--         * The study date.
+--     @studyDescription
+--         * The study description.
+--     @accessionNumber
+--         * The accession number associated for the study.
+--     @patientBirthDate
+--         * The patient's birth date.
+--
+-- RETURN VALUE
+--     None
+--
+CREATE OR ALTER PROCEDURE dbo.BulkUpdateStudyInstance
+    @batchSize                          INT,
+    @partitionKey                       INT,
+    @studyInstanceUid                   VARCHAR(64),
+    @patientId                          NVARCHAR(64) = NULL,
+    @patientName                        NVARCHAR(325) = NULL,
+    @referringPhysicianName             NVARCHAR(325) = NULL,
+    @studyDate                          DATE = NULL,
+    @studyDescription                   NVARCHAR(64) = NULL,
+    @accessionNumber                    NVARCHAR(64) = NULL,
+    @patientBirthDate                   DATE = NULL
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SET XACT_ABORT ON
+    BEGIN TRANSACTION
+
+    DECLARE @rowsUpdated INT = 0
+    DECLARE @imageResourceType AS TINYINT = 0
+    DECLARE @currentDate DATETIME2(7) = SYSUTCDATETIME()
+    DECLARE @updatedInstances AS TABLE
+           (PartitionKey INT,
+            StudyInstanceUid VARCHAR(64),
+            SeriesInstanceUid VARCHAR(64),
+            SopInstanceUid VARCHAR(64),
+            StudyKey BIGINT,
+            SeriesKey BIGINT,
+            InstanceKey BIGINT,
+            Watermark BIGINT)
+
+    DECLARE @totalCount INT = (SELECT COUNT(*) FROM dbo.Instance WHERE PartitionKey = @partitionKey AND StudyInstanceUid = @studyInstanceUid AND Status = 1 AND NewWatermark IS NOT NULL) 
+
+    WHILE (@rowsUpdated < @totalCount)
```

Comment:
This is a good thought, but you are not commiting and letting go of the locks, so the batch update is not fully of help right?

Reply:
Yes, I didnt commit because I want to update all the instances in a study. It is not helping too much but doesnt want to consume more resources.

Reply:
As per the conversation, removed the batching as it wasnt much useful and the records are going much less